Remove deprecated pre-commit script and adjust doc (#6455)
python/typeshed: CONTRIBUTING.md

## Preparing the environment

### Code away!

> This repository has a Continuous Integration (CI) running on the pull requests,
> which includes automatically adjusting code formatting (`black`, `isort`) and
> running the tests. It means you can ignore all this on your side, focus on the
> code and rely on the CI to fix everything, or point you to the places that need
> fixing.

Reviewer: ```suggestion
Typeshed runs continuous integration (CI) on all pull requests. This will
automatically fix formatting (using `black`, `isort`) and run tests.
It means you can ignore all local setup on your side, focus on the
code and rely on the CI to fix everything, or point you to the places that
need fixing.
```

[virtual environment](https://docs.python.org/3/tutorial/venv.html) and install
certain libraries typeshed uses to check stub files.

Follow platform-specific instructions below.

> The repository is equipped with a [`pre-commit`](https://pre-commit.com/)

Reviewer 2: It would be nicer to say `pre-commit.ci`, to make it clear that you don't have to run anything locally.

Author: Done

configuration file. If you want some checks to run automatically when you

> commit, you can install the hooks:

Reviewer 2: Why would anyone want this? It would only run black and isort, and those run automatically when you make a pull request anyway.

Author: While I understand this might not be everyone's choice, I appreciate that some people like to post clean commits. Also on a more pragmatic point of view, this avoids launching about 40 jobs to cancel them after a few seconds, which has an (albeit small, yet) existing environmental impact :)

But I understand your point, in that we can make it clearer that this step is 100% optional.

Author: Is it good for you now ?

Reviewer 2: Thanks! I like it now :)

[virtual environment](https://docs.python.org/3/tutorial/venv.html) and install
certain libraries typeshed uses to check stub files.

> Follow platform-specific instructions below.

Reviewer 3: ```suggestion
Note that this step is entirely optional, as your code will be
automatically reformatted when your PR is submitted
(see **Code Formatting** below),
and the full test suite will also be run automatically.

Follow platform-specific instructions below.
```

Author: I'll try to make a commit along those lines :)

Reviewer 3: I absolutely agree that running the tests locally is valuable! But in the **Contribution process at a glance** section, it says "Optionally run the tests". The wording here sounds much stronger at the moment -- I think the two sections should be consistent with each other 🙂 Also, I've often found typeshed's CI to be _far_ faster than my machine (but, maybe that's just a sign I need a new laptop...)

(.venv3) > python -m pip install -r requirements-tests-py3.txt
```

## Code formatting

The code is formatted by `black` and `isort`.

The repository is equipped with a [`pre-commit.ci`](https://pre-commit.ci/)
configuration file. This means that you don't *need* to do anything yourself to
run the code formatters. When you push a commit, a bot will run those for you

> right away and add a commit to your PR. Neat, no ?

Reviewer 2: Nit:
```suggestion
right away and add a commit to your PR. Neat, no?
```
